src/model.cpp

```diff
@@ -710,6 +761,21 @@ void flattenComponent(const ComponentEntityPtr &parent, const ComponentPtr &comp
         // Get list of required units from component's variables.
         std::vector<UnitsPtr> requiredUnits = unitsUsed(importedComponentCopy);
 
+        // Get the names of the units used by the <cn> elements in all the components' MathML and add them
+        // to the required units.
+        auto tempModel = Model::create();
+        tempModel->addComponent(importedComponentCopy);
+        tempModel->linkUnits();
+        auto cnList = cnUnits(tempModel);
+        for (auto &name : cnList) {
+            if (!tempModel->hasUnits(name)) {
+                auto u = importModel->units(name);
+                if (u != nullptr) {
+                    requiredUnits.push_back(u);
+                } // TODO else warning
```

Comment:
Comment can go.

Reply:
As soon as we have an Issue class we can add the warnings ... this is just a reminder of where they'll be needed.

Reply:
So that should be in an issue then, no need for it to be in the codebase.